components/partial/code/CodeExamples.tsx

```diff
@@ -0,0 +1,154 @@
+import styled from "@emotion/styled";
+import {
+  Children,
+  useContext,
+  cloneElement,
+  useMemo,
+  useState,
+  useEffect,
+} from "react";
+import sortBy from "lodash.sortby";
+import {
+  CODE_BLOCK_PURPLE,
+  CODE_BLOCK_AQUA,
+  GREY_3,
+  PURPLE_DARK,
+  PURPLE_DARKER,
+  GREY_5,
+  CODE_BLOCK_ORANGE,
+} from "../../colors";
+import { getLanguage } from "../../util/groupCodeExamples";
+import { LanguageContext } from "../../util/Contexts";
+import highlight from "../../../modules/highlight";
+import Select from "../../base/select/Select";
+
+const SwitcherBar = styled.div`
+  background-color: ${PURPLE_DARKER};
+  padding: 10px 8px;
+  border-top-left-radius: 5px;
+  border-top-right-radius: 5px;
+  display: flex;
+`;
+
+const CodeBlock = styled.div`
+  background-color: ${PURPLE_DARK};
+  overflow-x: scroll;
+  padding: 12px;
+  border-bottom-left-radius: 5px;
+  border-bottom-right-radius: 5px;
+
+  > pre {
+    margin: 0;
+
+    > code {
+      color: ${GREY_3};
+
+      .hljs- {
```

Comment:
Wasn't quite sure which colors to use for what, so these may be something to revisit.